src/main/groovy/liquibase/harness/compatibility/advanced/AdvancedTestHelper.groovy

```diff
@@ -0,0 +1,207 @@
+        }
+        return inputList
+    }
+
+    static validateSqlChangelog(String expectedSqlChangelog, String generatedSqlChangelog) {
+        String replacementRegexp = "--(.*?)\r?\n" //removes all sql comments starting from "--" till the end of line
+        String replacementRegexpNoEOL = "--(.*?)\$" //removes all sql comments starting from "--" till the end of file
+        String cleanExpectedChangelog = expectedSqlChangelog
+                .replaceAll(replacementRegexp, "")
+                .replaceAll(replacementRegexpNoEOL, "")
+                .trim()
+        String cleanGeneratedChangelog = generatedSqlChangelog
+                .replaceAll(replacementRegexp, "")
+                .replaceAll(replacementRegexpNoEOL, "")
+                .trim()
+        def message
+        if (cleanExpectedChangelog.equalsIgnoreCase(cleanGeneratedChangelog)) {
+            message = "GENERATED SQL CHANGELOG IS CORRECT"
+        } else {
+            message = "GENERATED SQL CHANGELOG: \n $cleanGeneratedChangelog \n DOES NOT MATCH EXPECTED SQL CHANGELOG: \n $cleanExpectedChangelog"
+        }
+        uiService.sendMessage(message)
+        assert cleanExpectedChangelog.equalsIgnoreCase(cleanGeneratedChangelog)
+    }
+
+    static String getShortDatabaseName(String dbName) {
+        switch (dbName) {
+            case "percona-xtradb-cluster":
+                return "mysql"
+            case "db2-luw":
+                return "db2"
+            default:
+                return dbName
+        }
+    }
+
+    static String removeSchemaNames(String generatedSql, Database database, String schemaName) {
+        if (database.getShortName().equals("sqlite")) {
+            return generatedSql.toLowerCase()
+        }
+        return generatedSql.toLowerCase().replace(schemaName + ".", "").replace("\"" + schemaName + "\".", "")
+    }
+
+    static String cleanDiff(String diff) {
+        String replacementRegexpRef = "Reference Database(.*?)\r?\n" //removes Reference Database diff line to generalize test data
+        String replacementRegexpComp = "Comparison Database(.*?)\r?\n" //removes Comparison Database diff line to generalize test data
+        String replacementRegexpWS = "\\s+" //removes whitespaces
+        return diff
+                .replaceAll(replacementRegexpRef, "")
+                .replaceAll(replacementRegexpComp, "")
+                .replaceAll(replacementRegexpWS, "")
+    }
+
+    static LinkedHashMap<String, String> configureChangelogMap(String basePath, String shortDbName) {
+        def map = new LinkedHashMap<String, String>()
+        map.put("xmlChangelog", basePath + ".xml")
+        map.put("sqlChangelog", basePath + ".$shortDbName" + ".sql")
+        map.put("ymlChangelog", basePath + ".yml")
+        map.put("jsonChangelog", basePath + ".json")
+        return map
+    }
+
+    static validateChangelog( String changelogFormat, String changelogContent, String verificationSql, String change, String changeReversed) {
```

Comment:
Please create two separate methods instead of this one, name the first of them 'validateGeneratedChangelog' and the second one 'validateDiffChangelog'